lib/bindings/http/http_emitter.js

```diff
@@ -0,0 +1,56 @@
+const BinaryHTTPEmitter = require("./emitter_binary.js");
+const StructuredEmitter = require("./emitter_structured.js");
+
+const {
+  SPEC_V03,
+  SPEC_V1
+} = require("./constants");
+
+/**
+ * A class which is capable of sending binary and structured events using
+ * the CloudEvents HTTP Protocol Binding specification.
+ *
+ * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md
+ * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#13-content-modes
+ */
+class HTTPEmitter {
+  /**
+   * Creates a new instance of {HTTPEmitter}. The default emitter uses the 1.0
+   * protocol specification in binary mode.
+   *
+   * @param {string} [version] The HTTP binding specification version. Default: "1.0"
+   * @throws {TypeError} if no options.url is provided or an unknown specification version is provided.
+   */
+  constructor(version = SPEC_V1) {
+    if (version !== SPEC_V03 && version !== SPEC_V1) {
+      throw new TypeError(
+        `Unknown CloudEvent specification version: ${version}`);
+    }
+    this.binary = new BinaryHTTPEmitter(version);
+    this.structured = new StructuredEmitter();
+  }
+
+  /**
+   * Sends the {CloudEvent} to an event receiver over HTTP POST
+   *
+   * @param {Object} options The configuration options for this event. Options
+   * provided will be passed along to Node.js `http.request()`.
+   * https://nodejs.org/api/http.html#http_http_request_options_callback
+   * @param {URL} options.url The HTTP/S url that should receive this event
+   * @param {CloudEvent} event the CloudEvent to be sent
+   * @param {string} [mode] the message mode for sending this event.
+   * Possible values are "binary" and "structured". Default: structured
+   * @returns {Promise} Promise with an eventual response from the receiver
+   */
+  send(options, event, mode = "binary") {
```

Comment:
@danbev any thoughts on this as a function signature? I'm not sure whether it makes sense to have an emitter always emit in binary or structured, or instead have the `mode` parameter in the ctor and once constructed it only emits in a single mode. Also... I don't love the whole `Constants` construct, but since it's everywhere, do you think we should have constants for `"binary"` and `"structured"`? 

Also... the first parameter is an `options` object, but the only required property is `url`. I had considered making the URL a string parameter, but we still need to accept an options object because this allows callers to pass things like `HttpAgent` here, which eventually makes its way to `http.request()`.

Reply:
> I'm not sure whether it makes sense to have an emitter always emit in binary or structured

My first thought was that it makes sense but thinking about this a little more I'm not so sure. For example, If I'm using this HttpEmitter then I know the destination of these events, the url we are posting to. If this destination is some kind of message broker I might be able to chose the best format for it (even though if can/must handle both it might always normalize into a specific content format). In that case I'm thinking just using one mode would be enough. Is my reasoning sane here?

 >I don't love the whole Constants construct, but since it's everywhere, do you think we should have constants for "binary" and "structured"?

Yeah, I think we should and then later perhaps look at a better solution for the constants.

Reply:
Thanks for the feedback @danbev. I've started [a thread](https://cloud-native.slack.com/archives/CCXF3CBL1/p1589377588205000) in the CE slack looking for feedback/opinions. If you have any further thoughts on this, we can hash it out there.

Will add "binary" and "structured" to the constants module. Will probably pull these up to be more accessible via the top level import. E.g.